src/libstd/io/mod.rs

```diff
@@ -322,6 +576,17 @@ pub trait Write {
     ///
     /// It is **not** considered an error if the entire buffer could not be
     /// written to this writer.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::io::Write;
+    ///
+    /// let mut buffer = Vec::new();
```

Comment:
Was it intentional that the reading examples worked with files and these work with vectors? It may be best to show off I/O going on here instead of in-memory buffers (as that's the common case).